src/Components/Web.JS/src/Platform/WebAssemblyProgressReporter.ts

```diff
@@ -0,0 +1,20 @@
+import { Blazor } from "../GlobalExports";
+import { WebAssemblyProgressService } from "./WebAssemblyProgressService";
+
+export class WebAssemblyProgressReporter {
+    static setProgress(resourcesTotal: number, resourcesLoaded: number) {
+        const circle = document.querySelector('.progress') as SVGCircleElement;
+        const circumference = 2 * Math.PI * circle.r.baseVal.value;
+        const progressPercentage = resourcesLoaded / resourcesTotal;
+        const ring = (1 - progressPercentage) * circumference;        circle.style.strokeDasharray = circumference.toString() + " " + circumference.toString();
+        circle.style.strokeDashoffset = ring.toString();
+        circle.style.display = 'block';
+        const element = document.getElementById('percentage') as unknown as SVGTextElement;
```

Comment:
Ah I see what you were doing in https://github.com/dotnet/aspnetcore/pull/42548#discussion_r912291346 now.

Reply:
Yes, it is cast to `unknown` and an error is produced since the type is not known initially. I did this in another part of the code that is dealing with SVG elements. I believe without the `unknown`, TS is assuming its type to be `HTMLCollectionOf<Element>` which cannot be converted to an SVG element.